backend/functions/src/v2/course/survey/functions.ts

```diff
@@ -0,0 +1,89 @@
+import { Request, Response } from "express";
+import { SurveyQuestion, IIndex } from "../../../common/types";
+import { getCourseRefByDocId } from "../../../common/utils";
+
+export const getSurvey = async (req: Request, res: Response) => {
+	const { courseId, studentId } = req.params;
+
+	const courseDocRef = (await getCourseRefByDocId(courseId).catch((err) => {
+		return res.status(404).send(err.toString());
+	})) as FirebaseFirestore.DocumentReference;
+	const surveyColRef = courseDocRef.collection("survey");
+
+	const surveyQuery = await surveyColRef
+		.where("studentId", "==", studentId)
+		.limit(1)
+		.get();
+
+	if (surveyQuery.empty) {
+		return res
+			.status(404)
+			.send(`Survey for ${studentId} doesn't exist under course ${courseId}`);
+	}
+	return res.status(200).send(surveyQuery.docs[0].data());
+};
+
+export const postSurvey = async (req: Request, res: Response) => {
+	const courseId = req.params.courseId;
+	const { studentId, fullName, surveyResponse } = req.body;
+	// Check essential fields in request body
+	const isInvalidBody = [courseId, studentId, fullName, surveyResponse].some(
```

Comment:
is this the newest one? coz I think we got rid of "courseId" right? theres also the question of whether it is to be "studentId" or "email"